platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/LatestEventTipsetTracker.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.gossip.shadowgraph;
+
+import com.swirlds.base.time.Time;
+import com.swirlds.common.platform.NodeId;
+import com.swirlds.platform.event.creation.tipset.Tipset;
+import com.swirlds.platform.event.creation.tipset.TipsetTracker;
+import com.swirlds.platform.internal.EventImpl;
+import com.swirlds.platform.system.address.AddressBook;
+import com.swirlds.platform.system.events.EventDescriptor;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Tracks the tipset of the latest self event. Must be thread safe, gossip will access it from multiple threads.
+ */
+public class LatestEventTipsetTracker { // TODO add a unit test
```

Comment:
TODO in code.

Reply:
Lol, I removed this two minutes after the original push. Thanks for the prompt review, I guess! 😺